src/engine/source/hlp/src/parsers/parse_field.cpp

```diff
@@ -86,6 +86,19 @@ void unescape(bool is_escaped, std::string& vs, std::string_view escape)
     }
 }
 
+std::string convertDotToSlash(std::string_view key)
+{
+    std::string result;
+    result.reserve(key.size());
+
+    for (char c : key)
+    {
+        result += (c == '.') ? '/' : c;
+    }
+
+    return result;
```

Comment:
Most of the time you will not find dots. Copying character by character is expensive. 
Use the STL, is faster.

```suggestion
    auto pos = key.find('.');
    if (pos == std::string_view::npos) {
        return std::string{key};
    }

    std::string s{key};
    std::replace(s.begin() + pos, s.end(), '.', '/');
    return s;
```